### PR TITLE
Debugmapcmd

### PR DESCRIPTION
**Pull request by AJenbo**
- Clean up god mode (the debug command was actually switching god+tools)
- Fix changing to regular levels when in a quest map
- Add command for loading quest maps
- Remove arguments that are now available as commands
- Remove chat buttons, it was largely hardcoded to specific keys and not easy to understand or remember

### CODE REVIEW

File: Source/debug.cpp [hunk 1]
```diff
@@ -101,13 +106,41 @@ std::string DebugCmdHelp(const std::string_view parameter)
...
 	return "You are now rich! If only this was as easy in real life...";
 }
 
 std::string DebugCmdTakeGoldCheat(const std::string_view parameter)
 {
-	TakeGoldCheat();
+	auto &myPlayer = Players[MyPlayerId];
+
+	for (auto itemId : myPlayer.InvGrid) {
+		itemId -= 1;
```
Comment (qndel) [inline Source/debug.cpp:132]: how is this supposed to work?

File: Source/debug.cpp [hunk 2]
```diff
@@ -137,106 +194,54 @@ std::string DebugCmdResetLevel(const std::string_view parameter)
...
 }
 
-std::string DebugCmdMaxSpellLevel(const std::string_view parameter)
+std::string DebugCmdVision(const std::string_view parameter)
 {
-	MaxSpellsCheat();
-	return "Knowledge is power.";
+	DebugVision = !DebugVision;
+	if (DebugVision)
+		return "You see as i do.";
```
Comment (qndel) [inline Source/debug.cpp:214]:
```suggestion
		return "You see as I do.";

```

File: Source/missiles.cpp [hunk 3]
```diff
@@ -264,13 +267,12 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, int t, bool
...
 		return ret;
 
+	if (hit >= hper) {
 #ifdef _DEBUG
-	if (hit >= hper && !debug_mode_key_inverted_v && !debug_mode_dollar_sign)
-		return false;
-#else
-	if (hit >= hper)
-		return false;
+		if (!DebugGodMode)
```
Comment (qndel) [inline Source/missiles.cpp:272]: I hate this identation so much, can't we deal with it somehow? 😅

File: Source/debug.cpp [hunk 4]
```diff
@@ -260,6 +265,50 @@ void SetAllSpellsCheat()
...
+
+	return "Knowledge is power.";
+}
+
+std::vector<DebugCmdItem> DebugCmdList = {
+	{ "help", "Prints help overview or help for a specific command.", "({command})", &DebugCmdHelp },
+	{ "give gold", "Fills the inventory with gold.", "", &DebugCmdGiveGoldCheat },
+	{ "give xp", "Levels the player up (min 1 level or {levels}).", "({levels})", &DebugCmdLevelUp },
+	{ "give spells", "Add all spells to player.", "", &DebugCmdGetAllSpells },
+	{ "give spells 10", "Set spell level to 10 for all spells.", "", &DebugCmdMaxSpellLevel },
```
Comment (qndel) [inline Source/debug.cpp:291]: give spells takes priority over give spells 10, so it's impossible to execute give spells 10, merge these 2 commands into one that takes spell level as an argument

File: Source/missiles.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/missiles.cpp:272]: So do I, but it's a step up from the duplicate

File: Source/debug.cpp (hunk 4 above)
Comment (AJenbo) [reply Source/debug.cpp:291]: Good point, but it's like that currently so shouldn't stop this pr

File: Source/debug.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/debug.cpp:132]: Idk, it was like this when I found it, just moved the code

**Review by qndel — APPROVED** [review]
👍

**Comment by obligaron** [comment]
Great to see more debug commands and the cleanup to the existing logic is nice. 👍 🙂

**Pull request merged**